pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -366,9 +366,32 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 		}
 	}
 
+	// failover mons running on host path to use persistent volumes if VolumeClaimTemplate is set and vice versa
+	for _, mon := range c.ClusterInfo.Monitors {
+		if c.HasMonPathChanged(mon.Name) {
+			logger.Infof("fail over mon %q due to change in mon path", mon.Name)
+			c.failMon(len(c.ClusterInfo.Monitors), desiredMonCount, mon.Name)
+			return nil
+		}
+	}
+
 	return nil
 }
 
+// HasMonPathChanged checks if the mon storage path has changed from host path to persistent volume or vice versa
+func (c *Cluster) HasMonPathChanged(mon string) bool {
```

Comment:
How about adding a unit test for this method?

Reply:
added unit test for this method.